src/nvim/api/vim.c

```diff
@@ -2116,27 +2116,27 @@ Dictionary nvim_eval_statusline(String str, Dict(eval_statusline) *opts, Error *
   }
   if (HAS_KEY(opts->use_winbar)) {
     use_winbar = api_object_to_bool(opts->use_winbar, "use_winbar", false, err);
-    use_bools++;
     if (ERROR_SET(err)) {
       return result;
     }
+    use_bools++;
   }
   if (HAS_KEY(opts->use_tabline)) {
     use_tabline = api_object_to_bool(opts->use_tabline, "use_tabline", false, err);
-    use_bools++;
     if (ERROR_SET(err)) {
       return result;
     }
-  }
-  if (HAS_KEY(opts->use_statuscol)) {
-    use_statuscol = api_object_to_bool(opts->use_statuscol, "use_statuscol", false, err);
     use_bools++;
-    if (ERROR_SET(err)) {
+  }
+  if (HAS_KEY(opts->statuscol_lnum)) {
+    VALIDATE_T("statuscol_lnum", kObjectTypeInteger, opts->statuscol_lnum.type, {
       return result;
-    }
+    });
+    statuscol_lnum = (int)opts->statuscol_lnum.data.integer;
```

Comment:
we should probably add some sort of bounds check to `statuscol_lnum`

Reply:
I don't think so, evaluating statuscolumn for a line number out of bounds of the actual buffer lines doesn't hurt anyone, and may be useful?

Reply:
Although negative numbers don't make sense and would break the current logic, I'll add that.

Reply:
> Although negative numbers don't make sense, we could filter those.

That is what I meant. Also `statuscol_lnum` is used as an argument to some functions which might throw an error for invalid line numbers, though I'm not sure if that's the case.

Reply:
Yep I added the negative range check. My first instinct was just to not set an upper bound, as evaluating for line numbers greater than `w_buffer->b_ml.ml_line_count` may be useful.

Reply:
> Yep I added the negative range check. My first instinct was just to not set an upper bound, as evaluating for line numbers greater than `w_buffer->b_ml.ml_line_count` may be useful.
> 
> EDIT: luckily, it seems the relevant functions don't fail, or fail silently when passing in a number greater than `w_buffer->b_ml.ml_line_count`.

But the statusline string itself might need to use contents of the line, in which case it would throw an error for an invalid line number. One could argue that that's the user's fault for passing both an invalid line number and a string that attempts to get contents from that line, but stuff like that can be hard to debug for a user, it's best to just not allow it imo.

Reply:
Yeah maybe